app/scripts/modules/core/src/managed/constraints/ConstraintCard.tsx

```diff
@@ -22,26 +27,93 @@ const logUnsupportedConstraintError = (type: string) => {
   );
 };
 
+const overrideConstraintStatus = (
+  application: Application,
+  options: Omit<IUpdateConstraintStatusRequest, 'application'>,
+) =>
+  ManagedWriter.updateConstraintStatus({
+    application: application.name,
+    ...options,
+  }).then(() => {
+    const dataSource: ApplicationDataSource<IManagedApplicationEnvironmentSummary> = application.getDataSource(
+      'environments',
+    );
+
+    // Here we wait to say things are fully done until we attempt to refresh, but don't
+    // reject. If the refresh fails things are going to be in a bad state anyway,
+    // so we don't want to wrongly imply that the override didn't take effect
+    // just because we're unable check.
+    return dataSource.refresh().catch(() => null);
+  });
```

Comment:
I am open to feedback re: this approach of dereferencing the data source off the app model all the way down here in this child component. It seemed really awkward and mis-aligned to put all this request/state logic _entirely_ up in `ArtifactDetail`, but specifically calling data sources in here felt a little off. I was considering passing down the `refresh` function from `useDataSource` via props into this function, but it does also involve threading another prop all the way down the tree and into these callbacks. Thoughts?

Reply:
One approach I've used in the past is the `useReducer` model and let a high-level parent component take control of such data needs. But that would involve quite a bit of re-write of this component I guess (like the pending status etc)?